src/similarity.spec.ts

```diff
@@ -0,0 +1,333 @@
+import fs from "fs";
+import path from "path";
+import { computeSimilarity } from "./similarity";
+
+const DUMMY_RESPONSE = {
+  headers: {},
+  body: Buffer.from([]),
+  status: {
+    code: 200
+  }
+};
+
+describe("similarity", () => {
+  it("detects exact matches", () => {
+    expect(
+      computeSimilarity("POST", "/test", {}, Buffer.from([]), {
+        request: {
+          method: "POST",
+          path: "/test",
+          headers: {},
+          body: Buffer.from([])
+        },
+        response: DUMMY_RESPONSE
+      })
+    ).toBe(0);
+  });
+
+  it("excludes mismatching methods", () => {
+    expect(
+      computeSimilarity("GET", "/test", {}, Buffer.from([]), {
+        request: {
+          method: "POST",
+          path: "/test",
+          headers: {},
+          body: Buffer.from([])
+        },
+        response: DUMMY_RESPONSE
+      })
+    ).toBe(Infinity);
+  });
+
+  it("excludes mismatching paths", () => {
+    expect(
+      computeSimilarity("POST", "/test", {}, Buffer.from([]), {
+        request: {
+          method: "POST",
+          path: "/test/other",
+          headers: {},
+          body: Buffer.from([])
+        },
+        response: DUMMY_RESPONSE
+      })
+    ).toBe(Infinity);
+  });
+
+  it("counts query parameters differences", () => {
+    expect(
+      computeSimilarity("POST", "/test", {}, Buffer.from([]), {
+        request: {
+          method: "POST",
+          path: "/test?c=d",
+          headers: {},
+          body: Buffer.from([])
+        },
+        response: DUMMY_RESPONSE
+      })
+    ).toBe(2);
+    expect(
+      computeSimilarity("POST", "/test?a=b", {}, Buffer.from([]), {
+        request: {
+          method: "POST",
+          path: "/test",
+          headers: {},
+          body: Buffer.from([])
+        },
+        response: DUMMY_RESPONSE
+      })
+    ).toBe(2);
+    expect(
+      computeSimilarity("POST", "/test?a=b", {}, Buffer.from([]), {
```

Comment:
Is it worth testing
- `/test?a=b&c=d` vs. `/test?a=b&c=f` <= 1 same, 1 different
- `/test?a=b&c=d` vs. `/test?a=b&c=d` <= both are the same

Reply:
Nice catch, this actually caught a bug :)